astropy/io/fits/connect.py

```diff
@@ -162,6 +163,9 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
+    # Initial `Time` state information
+    time_state = None
```

Comment:
Easier to always create the `FITS_Time()` object here.  It's very fast and cheap so better to let the code below be simplified.